Sources/xcprojectlint-package/NoDanglingFiles.swift

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2018 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import Foundation
+
+public func noDanglingFiles(_ project: Project, errorReporter: ErrorReporter) -> Int32 {
+    var result = EX_DATAERR
+
+    let targetFiles = Set(project.buildFiles.map{$1.fileRef})
+    let extensionsWhiteList = Set(["m", "mm", "swift", "cpp", "c"])
+    let actualFilesInProjectNavigator = project.fileReferences
+        .filter{ key, value in
+            let fileComponents = value.title.components(separatedBy: ".")
```

Comment:
I have a suggestion:
```
let results = project.fileReferences
	.flatMap{ _, fileRef -> String? in
	      let ext = fileRef.title.components(separatedBy: ".").last!
	      let shouldReport = extensionsWhiteList.contains(ext) && !targetFiles.contains(fileRef.id)
              if shouldReport {
                   return "\(project.absolutePathToReference(fileRef)):0: \(errorReporter.reportKind.logEntry) \(fileRef.path) is not added to any target.\n"
              } else {
                  return nil
              }
}
```
`let ext = fileRef.title.components(separatedBy: ".").last!` is fine, because title is `non-optional`:
`"".components(separatedBy: ".").last! // ""`
`"hello".components(separatedBy: ".").last! // "hello"`
`"hello.swift".components(separatedBy: ".").last! // "swift"`

Also, using `flatMap` instead of `filter` + `map` reduces computation complexity by `n` in worst case.